frontend/sozisel-app/src/components/ParticipantActiveSession/Modules/PollEvent/ParticipantPollEvent.tsx

```diff
@@ -0,0 +1,100 @@
+import "./ParticipantPollEvent.scss";
+import { Button, List, Paper, Typography } from "@material-ui/core";
+import {
+  ParticipantEvent,
+  Poll,
+  PollOption,
+  useSubmitPollResultsMutation,
+} from "../../../../graphql";
+import { ReactElement, useState } from "react";
+import PollEventOption from "./PollEventOption/PollEventOption";
+import PollIcon from "@material-ui/icons/Poll";
+import useCountdownTimer from "../../../../hooks/useCountdownTimer";
+import { useTranslation } from "react-i18next";
+
+export interface ParticipantPollEventProps {
+  token: string;
+  event: ParticipantEvent;
+  onPollFinished: () => void;
+}
+
+const ParticipantPollEvent = ({
+  token,
+  event,
+  onPollFinished,
+}: ParticipantPollEventProps): ReactElement => {
+  const { t } = useTranslation("common");
+  const pollData: Poll = event.eventData as Poll;
+  const [choosenOptionsIds, setChoosenOptionsIds] = useState<string[]>([]);
+  const [submitPollResultsMutation] = useSubmitPollResultsMutation();
+  const selectOption = (option: PollOption): void => {
+    if (!pollData.isMultiChoice) {
+      setChoosenOptionsIds([option.id]);
+      return;
+    }
+    if (choosenOptionsIds.some((id) => option.id === id)) {
+      setChoosenOptionsIds(choosenOptionsIds.filter((id) => option.id !== id));
```

Comment:
If you use the previous state value in `setState` I think it's better to pass function not value to `setState`, I mean:
```js
setChoosenOptionsIds(prev => prev.filter(id => option.id !== id))
```

Reply:
It does not matter really. This syntax you have shown is cool when you are inside some hook so that you don't have to pass the argument to a dependency list, otherwise its the same. In your case is is shorter though due to `prev` name and not `chosenOptionsIds` so I agree that @skusnierz can change it here.

Reply:
I changed it.